src/ray/core_worker/store_provider/memory_store/memory_store.h

```diff
@@ -0,0 +1,41 @@
+#ifndef RAY_CORE_WORKER_MEMORY_STORE_H
+#define RAY_CORE_WORKER_MEMORY_STORE_H
+
+#include "ray/common/id.h"
+#include "ray/common/status.h"
+#include "ray/core_worker/common.h"
+#include "ray/core_worker/store_provider/store_provider.h"
+
+namespace ray {
+
+class GetRequest;
+class CoreWorkerMemoryStore;
+
+/// The class provides implementations for local process memory store.
+/// An example usage for this is to retrieve the returned objects from direct
+/// actor call (see direct_actor_transport.cc).
+class CoreWorkerMemoryStore {
+ public:
+  CoreWorkerMemoryStore();
+  ~CoreWorkerMemoryStore(){};
+
+  Status Put(const RayObject &object, const ObjectID &object_id);
+  Status Get(const std::vector<ObjectID> &ids, int64_t timeout_ms,
+             std::vector<std::shared_ptr<RayObject>> *results, bool should_remove);
```

Comment:
1) `should_remove` -> `remove_after_get`
2) output arg `results` should be the last.
3) document these 3 methods

Reply:
makes sense.